waku/v2/protocol/waku_filter/waku_filter.nim

```diff
@@ -187,44 +188,59 @@ method init*(wf: WakuFilter) =
 
   wf.handler = handle
   wf.codec = WakuFilterCodec
+  wf.timeout = filterTimeout
 
-proc init*(T: type WakuFilter, peerManager: PeerManager, rng: ref BrHmacDrbgContext, handler: MessagePushHandler): T =
+proc init*(T: type WakuFilter, peerManager: PeerManager, rng: ref BrHmacDrbgContext, handler: MessagePushHandler,timeout: float = WakuFilterTimeout): T =
   let rng = crypto.newRng()
   var wf = WakuFilter(rng: rng,
                       peerManager: peerManager, 
                       pushHandler: handler)
-  wf.init()
-  
+  wf.init(filterTimeout = timeout)
   return wf
 
 proc setPeer*(wf: WakuFilter, peer: RemotePeerInfo) =
   wf.peerManager.addPeer(peer, WakuFilterCodec)
   waku_filter_peers.inc()
 
+proc handleClientError*(wf: WakuFilter, subscriber: Subscriber){.raises: [Defect, KeyError], async.} = 
+  var stringkey: string = $(subscriber)
+  if not wf.failedPeers.hasKey(stringkey):
+    wf.failedPeers[stringkey] = epochTime()
+  else:
+    var elapsedTime = epochTime() - wf.failedPeers[stringkey]
+    wf.failedPeers.del(stringkey)
+    if(elapsedTime > wf.timeout):
+      var index = wf.subscribers.find(subscriber)
+      wf.subscribers.delete(index)
+  return
+
 
 proc handleMessage*(wf: WakuFilter, topic: string, msg: WakuMessage) {.async.} =
   # Handle WakuMessage according to filter protocol
   trace "handle message in WakuFilter", topic=topic, msg=msg
-
+  var handleMessageFailed = false
+  var failedSubscriber: Subscriber

```

Comment:
Why do we need this intermittent variable if we are just going to pass subscriber here?

Reply:
We are changing the subscriber sequence , nim doesn't allow the same while iterating over the same.